foundry/foundry.py

```diff
@@ -185,7 +184,9 @@ def load(self, name, download=True, globus=True, verbose=False, metadata=None, a
 
         # unpack res, handle if empty
         try:
-            # if search returns multiple results, this automatically uses first result
+            # if search returns multiple results, this automatically uses first result, while warning the user
+            if len(res) > 1:
+                warnings.warn("Multiple datasets found for the given search query")
             res = res[0]
         except IndexError as e:
```

Comment:
An issue that looks like it predates your PR: Using try/except is an excessive route for testing whether a list is empty. 

Since you are already adding a "if len" test, could you convert this to statement to have 

```python
if len(res) == 0:
   raise Exception("load: No metadata found for dataset \"{name\"") 
```

Note that I also put the name of the dataset into the exception.

Reply:
@Aadit-Ambadkar once this fix is added we can merge! Everything else looks good.